### ISSUE TITLE
Incorrect ImageFormat cause wrong render result.

### ISSUE THREAD

**Issue by xVanTuring** [ajrcarey/pdfium-render#119]
When I try to render some colored pdf, I found that the color was wrong, the red turns to blue. After some debug, I found that the `PdfBitmap::as_bytes` actually outputs a `RGBA` sequence, but the `PdfBitmap::format` say it's `BGRA`, then in the `PdfBitmap::as_image` it swaps red and blue.
I'm not sure why this happens, but I can't reproduce it in WASM environment. \
The Pdf can be download from [here](https://github.com/dotnet-architecture/eBooks/blob/main/current/microservices/NET-Microservices-Architecture-for-Containerized-NET-Applications.pdf) in github.

## output & expected
![output](https://github.com/ajrcarey/pdfium-render/assets/11146248/f81c822c-1490-4ca4-a7c2-b6c76ed84fbf)
![expected](https://github.com/ajrcarey/pdfium-render/assets/11146248/bfc749c2-dd49-44d3-9bbd-d1d1b7c845a3)

## Code
``` rust
use pdfium_render::prelude::*;

pub fn main() -> Result<(), PdfiumError> {
    let bindings = Pdfium::bind_to_library(Pdfium::pdfium_platform_library_name_at_path("./"))
        .or_else(|_| Pdfium::bind_to_system_library())?;

    let pdfidum = Pdfium::new(bindings);
    let document = pdfidum.load_pdf_from_file(
        "F:/archive/pdf/NET-Microservices-Architecture-for-Containerized-NET-Applications.pdf",
        None,
    )?;

    let page = document.pages().get(0).unwrap();
    let bitmap = page.render_with_config(
        &PdfRenderConfig::new()
            .set_target_width(500)
            .set_maximum_height(500),
    )?;
    println!("Bitmap Format {:?}", bitmap.format());
    bitmap
        .as_image()
        .as_rgba8()
        .ok_or(PdfiumError::ImageError)?
        .save_with_format(format!("page{}.jpeg", 0), image::ImageFormat::Jpeg)
        .map_err(|_| PdfiumError::ImageError)?;
    Ok(())
}

```

## Env
* OS: Windows 10
* Pdfium-render: 0.8.15
* rustc: 1.73
* pdfium-binary: [PDFium 121.0.6110.0](https://github.com/bblanchon/pdfium-binaries/releases/tag/chromium%2F6110)

**Comment by xVanTuring** [comment]
Update:
If I manually set `set_reverse_byte_order` to `false`, the output will be correct. I noticed the note about `do_set_flag_reverse_byte_order: true` need to be true for `image>=0.24`, but i'm currently using `image = "0.24.6"`.
```rust
&PdfRenderConfig::new()
    .set_target_width(width)
    .set_reverse_byte_order(false)
```

**Comment by xVanTuring** [comment]
Here is a pdf with some colored-text.
[colored.pdf](https://github.com/ajrcarey/pdfium-render/files/13291110/colored.pdf)

**Comment by HeavenVolkoff** [comment]
Just experienced this issue. Disabling `set_reverse_byte_order` fixed. I think the `as_image` function should check if the `reverse_byte_order` is enabled or not, and skip the BGR conversion if it is.

**Comment by ajrcarey** [comment]
Thanks for your patience. The fundamental problem appears to be that even when setting the `set_reverse_byte_order()` flag to `true`, Pdfium still indicates that the bitmap format is BGRA instead of RGBA. Technically it should be the latter because the byte order of the B and R channels has been reversed, but Pdfium appears to always report BGRA irrespective of the channel order.

I'm wondering if changing the default value of `set_reverse_byte_order()` to `false` is sufficient to correct the behaviour. In my testing, every colored PDF I rendered was affected by the problem. Do either of you have examples of PDFs that are _not_ affected by this bug?

**Comment by ajrcarey** [comment]
As to why the problem does not affect WASM: if you are using the PdfBitmap::as_image_data() function in your WASM code, then the color swapping functionality used in PdfBitmap::as_image() is skipped entirely. This is a bug in its own right; whether the images are colored correctly or not, WASM code and non-WASM code should give the same result!

**Comment by ajrcarey** [comment]
Extracted pixel data color normalization functionality out of `PdfBitmap::as_image()` into new `PdfBitmap::as_rgb_bytes()` function. Deprecated `PdfBitmap::as_bytes()` in favour of renamed `PdfBitmap::as_raw_bytes()`. Updated WASM-specific `PdfBitmap::as_image_data()` function to consume output of `PdfBitmap::_as_rgb_bytes()` rather than `PdfBitmap::as_raw_bytes()`. This ensures that color channel normalization behaviour is identical in both WASM and non-WASM builds.

As tempting as it would be to simply change the default of `PdfRenderConfig::set_reverse_byte_order()` from `true` to `false`, it does occur to me that this would have the side effect of altering the output of the WASM-specific `PdfBitmap::as_array()` function, requiring consumers to perform their own color channel normalization and thus defeating the purpose of the function (which is to avoid extra memory allocations). This means that it's likely necessary for each `PdfBitmap` to know if it was created from a rendering configuration where `set_reverse_byte_order()` was set to `true` or not.

**Comment by ajrcarey** [comment]
Added reverse byte order flag to `PdfBitmap`. Set flag from `PdfPage::render_into_bitmap_with_settings()`. Adjusted `PdfBitmap::as_rgba_bytes()` to take byte order flag into account when determining how to normalize pixel color channel data into RGBA. The result is that, irrespective of the setting of `set_reverse_byte_order()`, color output is always RGBA and output is consistent between WASM and non-WASM builds. Updated README. Ready to release as part of crate version 0.8.16, pending additional testing.

**Comment by ajrcarey** [comment]
Released as part of crate version 0.8.16.
